.github/workflows/gcp-dsh-testing.yaml

```diff
@@ -0,0 +1,208 @@
+        run: |
+          DEFAULT_SC=$(kubectl get sc -o jsonpath='{.items[?(@.metadata.annotations.storageclass\.kubernetes\.io/is-default-class=="true")].metadata.name}')
+          kubectl get sc "$DEFAULT_SC" -o yaml | \
+            sed "s/name: $DEFAULT_SC/name: gp2/" | \
+            sed "/^  uid:/d; /^  resourceVersion:/d; /^  creationTimestamp:/d" | \
+            kubectl apply -f -
+            
+          cd self-hosted/charts/dz-control-plane-deps
+          find values -type f -exec sed -i'.bak' "s/example\.com/$JOB_IDENTIFIER\.ci\.selfzero\.net/g" {} \; && find values -name "*.bak" -delete
+          make install
+
+      - name: Update values.yaml for dz-control-plane
+        env:
+          BACKEND_LICENSE_KEY: ${{ secrets.BACKEND_LICENSE_KEY }}
+        run: |
+          # setting credentials enable to false since we will explicitly feed the dockerhub creds to kubernetes api
+          # also setting image.pullsecrets to empty to make sure that each of the deployments dont try to pull their relevant OCI images from this registry
+          # backend license key is ... needed
+
+          yq e '.credentials.enable = false | .backend.licenseKey = strenv(BACKEND_LICENSE_KEY) | .image.pullSecrets = []' -i self-hosted/charts/dz-control-plane/values.yaml
+
+      - name: Deploy DevZero Control Plane (after configuring kubernetes to use dockerhub creds, and patching all the deployments to point to the right domain)
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+        run: |
+          cd self-hosted/charts/dz-control-plane
+          make add-docker-creds
+          find . -name "values.yaml" -exec sed -i'.bak' "s/example\.com/$JOB_IDENTIFIER\.ci\.selfzero\.net/g" {} \; && find . -name "values.yaml.bak" -delete
+          make install
+
+      - name: Validate Control Plane
+        run: |
+          echo -e "\nPods in namespace  devzero:"
+          kubectl get pods -n devzero
+          chmod +x self-hosted/.github/scripts/dsh-pod-test.sh
+          self-hosted/.github/scripts/dsh-pod-test.sh
+          echo -e "\nIngress in namespace  devzero:"
+          kubectl get ingress -n devzero
+      
+      - name: Deploy Data Plane Dependencies
+        run: |
+          cd self-hosted/charts/dz-data-plane-deps
+          find values -type f -exec sed -i'.bak' "s/example\.com/$JOB_IDENTIFIER\.ci\.selfzero\.net/g" {} \; && find values -name "*.bak" -delete
+          make install
+
+      - name: Deploy DevZero Data Plane
+        run: |
+          cd self-hosted/charts/dz-data-plane
+          find . -name "values.yaml" -exec sed -i'.bak' "s/example\.com/$JOB_IDENTIFIER\.ci\.selfzero\.net/g" {} \; && find . -name "values.yaml.bak" -delete
+          make install
+          kubectl delete runtimeclass kata
+          kubectl apply -f ../../terraform/examples/gcp/base-cluster/runtimeclass.yaml
```

Comment:
The kata daemonset already has the option to create the runtimeclass, we may not need this: https://github.com/kata-containers/kata-containers/blob/main/tools/packaging/kata-deploy/scripts/kata-deploy.sh#L43